module/spl/spl-rwlock.c

```diff
@@ -32,7 +32,32 @@
 
 #define DEBUG_SUBSYSTEM S_RWLOCK
 
-#if defined(CONFIG_RWSEM_GENERIC_SPINLOCK)
+#if defined(CONFIG_PREEMPT_RT_FULL)
+
+#include <linux/rtmutex.h>
+
+static int
+__rwsem_tryupgrade(struct rw_semaphore *rwsem)
+{
+	ASSERT(rt_mutex_owner(&rwsem->lock) != current);
```

Comment:
@clefru this puzzled me too, initially. But, wasn't the first statement right? You your'e doing upgrade, you are only a writer, hence not a owner.